File: backend/models/Participant.js
// models/Participant.js
const mongoose = require('mongoose');

const ParticipantSchema = new mongoose.Schema({
  name: { type: String, required: true },
  college: { type: String, required: true },
  email: { type: String },
  phone: { type: String },
  eventId: { type: String }, // if multiple events
  registeredAt: { type: Date, default: Date.now },
  registrationQRCode: { type: String }, // DataURI (png) or URL to file storage
  approved: { type: Boolean, default: false },
  entryPassQRCode: { type: String }, // DataURI or URL
});

module.exports = mongoose.model('Participant', ParticipantSchema);

File: java-backend/src/main/java/com/vamsi/event/model/Participant.java
// src/main/java/com/vamsi/event/model/Participant.java
package com.vamsi.event.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Data
@Document(collection = "participants")
public class Participant {
    @Id
    private String id;
    private String name;
    private String college;
    private String email;
    private String phone;
    private String eventId;
    private Instant registeredAt = Instant.now();
    private String registrationQRCode;
    private boolean approved = false;
    private String entryPassQRCode;
}